metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/WorkflowExecutionDao.java

```diff
@@ -132,13 +136,28 @@ public void updateMonitorInformation(WorkflowExecution workflowExecution) {
         updateResults == null ? 0 : updateResults.getUpdatedCount());
   }
 
-  public void setCancellingState(WorkflowExecution workflowExecution) {
+  /**
+   * Set the cancelling field in the database.
+   * <p>Also adds information of the user identifier that cancelled the execution or if it was by a
+   * system operation, using {@link CancelledSystemId} values as identifiers. For historical
+   * executions the value of the <code>cancelledBy</code> field will remain <code>null</code></p>
+   *
+   * @param workflowExecution the workflowExecution to be cancelled
+   * @param metisUser the user that triggered the cancellation or null if it was the system
+   */
+  public void setCancellingState(WorkflowExecution workflowExecution, MetisUser metisUser) {
     UpdateOperations<WorkflowExecution> workflowExecutionUpdateOperations = morphiaDatastoreProvider
         .getDatastore().createUpdateOperations(WorkflowExecution.class);
     Query<WorkflowExecution> query = morphiaDatastoreProvider.getDatastore()
         .find(WorkflowExecution.class)
         .filter("_id", workflowExecution.getId());
     workflowExecutionUpdateOperations.set("cancelling", Boolean.TRUE);
+    if (metisUser != null && metisUser.getUserId() != null) {
+      workflowExecutionUpdateOperations.set("cancelledBy", metisUser.getUserId());
```

Comment:
This is a tiny thing: maybe use the if statement only to determine the value of the user ID? Then you don't have the 'duplicate' part of setting the 'cancelledBy' parameter in the operations object (and you don't have to use the String constant twice).

Reply:
Fixed.